2.0: TypeError on IDP-initiated logout

**Describe the bug**
When trying to initiate a logout directly from the IDP a TypeError is thrown: `TypeError: Argument 2 passed to SimpleSAML\IdP\TraditionalLogoutHandler::startLogout() must be of the type string, null given`

**To Reproduce**
Steps to reproduce the behavior:
1. Go to `http://idp/simplesaml/module.php/saml/idp/initSingleLogout?RelayState=http://idp`
2. See error

**Expected behavior**
Expected to get logged out of all known SPs.

**Screenshots or logs**
```
SimpleSAML\Error\Error: UNHANDLEDEXCEPTION
Backtrace:
2 .../vendor/simplesamlphp/simplesamlphp/public/_include.php:31 (SimpleSAML_exception_handler)
1 .../vendor/symfony/error-handler/ErrorHandler.php:607 (Symfony\Component\ErrorHandler\ErrorHandler::handleException)
0 [builtin] (N/A)
Caused by: TypeError: Argument 2 passed to SimpleSAML\IdP\TraditionalLogoutHandler::startLogout() must be of the type string, null given
Backtrace:
4 .../vendor/simplesamlphp/simplesamlphp/src/SimpleSAML/IdP/TraditionalLogoutHandler.php:85 (SimpleSAML\IdP\TraditionalLogoutHandler::startLogout)
3 [builtin] (call_user_func_array)
2 .../vendor/simplesamlphp/simplesamlphp/src/SimpleSAML/HTTP/RunnableResponse.php:72 (SimpleSAML\HTTP\RunnableResponse::sendContent)
1 .../vendor/symfony/http-foundation/Response.php:394 (Symfony\Component\HttpFoundation\Response::send)
0 .../vendor/simplesamlphp/simplesamlphp/public/module.php:14 (N/A)
```

**Additional context**
There seems to be some confusion if `assocId` in `startLogout` is/should be nullable:
```
src/SimpleSAML/IdP/LogoutHandlerInterface.php
    public function startLogout(array &$state, string $assocId): void;
src/SimpleSAML/IdP/IFrameLogoutHandler.php
    public function startLogout(array &$state, ?string $assocId): void
src/SimpleSAML/IdP/TraditionalLogoutHandler.php
    public function startLogout(array &$state, string $assocId): void
modules/core/src/Controller/Logout.php
        $assocId = $state['core:TerminatedAssocId'];
        return new RunnableResponse([$idp->getLogoutHandler(), 'startLogout'], [&$state, $assocId]);
src/SimpleSAML/IdP.php
        if ($assocId !== null) {
            $handler = $this->getLogoutHandler();
            $handler->startLogout($state, $assocId);
```

Since this is IDP-initiated there's no association that initiated the logout, which in several places is accounted for. Looking at the code I suspect, that the interface for `startLogout` should get a nullable `assocId` as well as all implementations of it. Marking `TraditionalLogoutHandler::startLogout`'s `assocId` nullable allows for proper logout.

--- a/docs/simplesamlphp-changelog.md
+++ b/docs/simplesamlphp-changelog.md
@@ -13,6 +13,7 @@ Released 2023-03-10
 * Fix some issues with logout (#1780, #1785)
 * The `loginpage_links` functionality for authsources was restored and documented (#1770, #1773)
 * Several issues regarding the use of the back-button were fixed (#1720)
+* Fixed IdP-initiated SLO (#1776)
 * Many fixes in documentation
 * Fixed config/authsources.php.dist so you can just rename it for new deployments to get you started (#1771)
 * Fixed UTF-8 encoding for metadata output

--- a/docs/simplesamlphp-changelog.md
+++ b/docs/simplesamlphp-changelog.md
@@ -10,10 +10,9 @@ See the upgrade notes for specific information about upgrading.
 Released 2023-03-10
 
 * The language-menu on mobile devices was fixed
-* Fix some issues with logout (#1780, #1785)
+* Fix some issues with logout (#1776, #1780, #1785)
 * The `loginpage_links` functionality for authsources was restored and documented (#1770, #1773)
 * Several issues regarding the use of the back-button were fixed (#1720)
-* Fixed IdP-initiated SLO (#1776)
 * Many fixes in documentation
 * Fixed config/authsources.php.dist so you can just rename it for new deployments to get you started (#1771)
 * Fixed UTF-8 encoding for metadata output

--- a/src/SimpleSAML/IdP.php
+++ b/src/SimpleSAML/IdP.php
@@ -437,7 +437,7 @@ class IdP
                 throw new Error\Exception('Unknown logout handler: ' . var_export($logouttype, true));
         }
 
-        /** @var IdP\LogoutHandlerInterface */
+        /** @var \SimpleSAML\IdP\LogoutHandlerInterface */
         return new $handler($this);
     }
 

--- a/src/SimpleSAML/IdP/LogoutHandlerInterface.php
+++ b/src/SimpleSAML/IdP/LogoutHandlerInterface.php
@@ -29,9 +29,9 @@ interface LogoutHandlerInterface
      * This function must never return.
      *
      * @param array &$state The logout state.
-     * @param string $assocId The association that started the logout.
+     * @param string|null $assocId The association that started the logout.
      */
-    public function startLogout(array &$state, string $assocId): void;
+    public function startLogout(array &$state, ?string $assocId): void;
 
 
     /**

--- a/src/SimpleSAML/IdP/TraditionalLogoutHandler.php
+++ b/src/SimpleSAML/IdP/TraditionalLogoutHandler.php
@@ -80,9 +80,9 @@ class TraditionalLogoutHandler implements LogoutHandlerInterface
      * This function never returns.
      *
      * @param array  &$state The logout state.
-     * @param string $assocId The association that started the logout.
+     * @param string|null $assocId The association that started the logout.
      */
-    public function startLogout(array &$state, string $assocId): void
+    public function startLogout(array &$state, /** @scrutinizer ignore-unused */?string $assocId): void
     {
         $state['core:LogoutTraditional:Remaining'] = $this->idp->getAssociations();
 

